app/models/rule_group_relationship.rb

```diff
@@ -11,6 +11,6 @@ class RuleGroupRelationship < ApplicationRecord
   validates :left_id, presence: true, uniqueness: { scope: %i[relationship right_id right_type left_type] }
 
   scope :with_relationships, lambda { |rules_or_groups, relationship|
-    where(relationship: relationship).includes(:left, :right).where(left: rules_or_groups)
+    where(relationship: relationship, left: rules_or_groups).includes(:left, :right)
```

Comment:
Not sure about the `includes` here, especially if it does not have `references`.

Reply:
Yeah, I only added the `includes` in this scope because we're calling `left` and `right` here:
```
def map_rule_group_relationships(relationship)
      relationships_for(rules, rule_groups, relationship).each_with_object({}) do |rgr, relationships|
        relationships[rgr.left] ||= []
        relationships[rgr.left] << rgr.right
     end
end
```
But I could do `relationships_for(rules, rule_groups, relationship).includes(:left, :right)` so that the `includes` isn't inside the scope? Or I could add `references` after the `includes` in the scope?

Reply:
You can do two arguments to references, or no?

Reply:
I tried to add references but it doesn't work with polymorphic associations apparently:
`ActiveRecord::EagerLoadPolymorphicError: Cannot eagerly load the polymorphic association :left`

Reply:
Ok, let's keep it as it is...